datamations/datamation_groupby.py

```diff
@@ -59,24 +61,32 @@ def prep_specs_summarize(self, width=300, height=300):
         if len(self.states[1].groups.keys()) == 2:
             groups = list(self.states[1].groups.keys())
         else:
-            groups = [list(self.states[1].groups.keys())[0][0], list(self.states[1].groups.keys())[2][0]]
-            subgroups = [list(self.states[1].groups.keys())[0][1], list(self.states[1].groups.keys())[1][1]]
+            groups = list(sorted(set(map(lambda x: x[0], self.states[1].groups.keys()))))
+            subgroups = list(sorted(set(map(lambda x: x[1], self.states[1].groups.keys()))))
+            if (len(self._by) > 2):
+                l3groups = []
+                for key in list(map(lambda x: "NA" if pd.isna(x[2]) else str(x[2]), self.states[1].groups.keys())):
+                    if key not in l3groups:
+                        l3groups.append(key)
 
         specs_list = []
 
         labels = [subgroups[0] if len(self._by) > 1 else groups[0], subgroups[1] if len(self._by) > 1 else groups[1]]
 
+        if (len(self._by) > 2):
+            sublabel = "round(datum.label) == 1 ? '" + l3groups[0] + "' : " + "round(datum.label) == 2 ? '" + l3groups[1] + "' : '" + l3groups[2] + "'"
+
         x_encoding = {
             "field": "datamations_x",
             "type": "quantitative",
             "axis": {
-            "values": [1, 2],
-            "labelExpr": "round(datum.label) == 1 ? '" + labels[0] + "' : '"  +  labels[1] + "'",
+            "values": list(range(1, len(groups)+1)),
+            "labelExpr": sublabel if len(self._by) > 2 else "round(datum.label) == 1 ? '" + labels[0] + "' : '"  +  labels[1] + "'",
             "labelAngle": -90
             },
-            "title": self._by[1] if len(self._by) > 1 else x_axis,
+            "title": self._by[-1] if len(self._by) > 1 else x_axis,
             "scale": {
-            "domain": [0.5, 2.5]
+            "domain": [0.5, 3.5 if len(self._by) > 2 else 2.5]
```

Comment:
I believe on the R side of spec generation, we changed this to just always be [0, 3], which would position 2 axes values equidistant from the sides of container and each other. Not super important to change, but fixes some alignment issues at Giorgi's suggestion. You can see x domain in a few places here:

https://github.com/microsoft/datamations/blob/a4213904a70817da77014192c42300a129d1234b/sandbox/custom_animations/custom-animations-mean-R.json#L815

Reply:
Update here would also fix a failed test :) https://github.com/microsoft/datamations/issues/145#issuecomment-1083827203

Reply:
Done. thanks!